File: sprint1/By lessons/lesson10code1/lesson10code1.cpp
#include <iostream>
#include <vector>
using namespace std;

int solve(vector<int>& array) {
    int maximal_sum = 0;
    for (int i = 0; i < array.size(); ++i) {
        int current_sum = 0;
        for (int j = 0; j <= i; ++j) {
            current_sum += array[j];
        }
        if (maximal_sum < current_sum) {
            maximal_sum = current_sum;
        }
    }
    return maximal_sum;
}

File: sprint1/By lessons/lesson10code1/lesson10code1.cs
public static int Solve(List<int> array)
{
    int maximal_sum = 0; // Сумма пустого префикса равна 0

    for (int i = 0; i < array.Count; i++)
    {
        int current_sum = array.GetRange(0, i + 1).Sum();

        if (maximal_sum < current_sum)
        {
            maximal_sum = current_sum;
        }
    }

    return maximal_sum;
}
